src/app/meshes/components/MeshDetails.vue

```diff
@@ -0,0 +1,186 @@
+                  </router-link>
+                </li>
+              </ul>
+            </DefinitionListItem>
+          </DefinitionList>
+        </div>
+      </template>
+    </KCard>
+
+    <KCard>
+      <template #body>
+        <ResourceCodeBlock
+          id="code-block-mesh"
+          :resource-fetcher="fetchMesh"
+          :resource-fetcher-watch-key="mesh?.name || null"
+        />
+      </template>
+    </KCard>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { KBadge, KCard } from '@kong/kongponents'
+import { PropType, computed } from 'vue'
+import { useRoute } from 'vue-router'
+
+import MeshCharts from '../components/MeshCharts.vue'
+import DefinitionList from '@/app/common/DefinitionList.vue'
+import DefinitionListItem from '@/app/common/DefinitionListItem.vue'
+import ResourceCodeBlock from '@/app/common/ResourceCodeBlock.vue'
+import TextWithCopyButton from '@/app/common/TextWithCopyButton.vue'
+import { useStore } from '@/store/store'
+import type { SingleResourceParameters } from '@/types/api.d'
+import type { Mesh, MeshInsight } from '@/types/index.d'
+import { useKumaApi, useI18n } from '@/utilities'
+import { notEmpty } from '@/utilities/notEmpty'
+
+const { t, formatIsoDate } = useI18n()
+const kumaApi = useKumaApi()
+const route = useRoute()
+const store = useStore()
+
+const props = defineProps({
+  mesh: {
+    type: Object as PropType<Mesh>,
+    required: true,
+  },
+
+  meshInsight: {
+    type: Object as PropType<MeshInsight>,
+    required: true,
+  },
+})
+
+const basicMesh = computed(() => {
+  const { name, creationTime, modificationTime } = props.mesh
+  return {
+    name,
+    created: formatIsoDate(creationTime),
+    modified: formatIsoDate(modificationTime),
+    'Data Plane Proxies': props.meshInsight.dataplanes.total,
+  }
+})
+
+const extendedMesh = computed(() => {
+  const mtls = getBackendData(props.mesh, 'mtls')
+  const logging = getBackendData(props.mesh, 'logging')
+  const metrics = getBackendData(props.mesh, 'metrics')
+  const tracing = getBackendData(props.mesh, 'tracing')
+  const localityAwareLoadBalancing = Boolean(props.mesh.routing?.localityAwareLoadBalancing)
+
+  return {
+    mtls,
+    logging,
+    metrics,
+    tracing,
+    localityAwareLoadBalancing,
+  }
+})
+
+const totalPolicyCount = computed(() => {
+  return Object.values(props.meshInsight.policies ?? {}).reduce((total, stat) => total + stat.total, 0)
+})
+const policyTypes = computed(() => {
+  return Object.entries(props.meshInsight.policies ?? {})
+    .map(([policyTypeName, stat]) => {
+      const policyType = store.state.policyTypesByName[policyTypeName]
+
+      if (policyType && stat.total !== 0) {
+        return {
+          name: policyType.name,
+          path: policyType.path,
+          total: stat.total,
+        }
+      }
+
+      return null
+    })
+    .filter(notEmpty)
+})
+
+function getBackendData(mesh: Mesh, field: 'mtls' | 'logging' | 'metrics' | 'tracing'): string | boolean {
```

Comment:
isn't `string or boolean` a bit weird of a type?

Reply:
You’re right. I’ve rewrote this a bit here: https://github.com/kumahq/kuma-gui/pull/1286/commits/37f0444d9b01c967e72e0bc2ce0aeba3b336c5b1